### PR TITLE
[router] Create enum type for router resources

### PR DESCRIPTION
**Pull request by majisourav99**
## [router] Create enum type for router resources

The router resource type was used as a raw string and compared in hot path in various request handler. This PR creates an enum class for various router resource types for a cleaner and better usage in various handler classes. 

## How was this PR tested?

unit/CI tests
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java [hunk 1]
```diff
@@ -109,7 +110,7 @@ public String getKey() {
   }
 
   public boolean isInvalidStorageRequest() {
-    return StringUtils.isEmpty(resourceType) || (!resourceType.equals(TYPE_STORAGE))
+    return resourceType == RouterResourceType.TYPE_INVALID || resourceType != RouterResourceType.TYPE_STORAGE
```
Comment (FelixGV) [inline services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java:113]: The first condition is redundant, isn't it? `resourceType != RouterResourceType.TYPE_STORAGE` encompasses `resourceType == RouterResourceType.TYPE_INVALID` (and many others)?

**Review by FelixGV — COMMENTED** [review]
Thanks, LGTM. Just left a minor comment.

File: services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java (hunk 1 above)
Comment (majisourav99) [reply services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java:113]: Right! will remove it.

Comment (majisourav99) [reply services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java]: actually just got rid of it.  it was a useless test only method

Comment (FelixGV) [reply services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java]: Thanks! Do we have some static analysis issue?

Comment (majisourav99) [reply services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java]: dont think so, locally it runs fine but fails on GH. it says some sort of timeout

Comment (FelixGV) [reply services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java]: Let's figure it out...

Comment (FelixGV) [reply services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java]: Weird... that means static analysis is flaky? That's concerning... was it GitHub Actions instability?

Comment (majisourav99) [reply services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java]: yeah, there was one type check issue, but even after resolving that, it still failed quite a few times.

**Pull request merged**